internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_inventory_items_data_source.go

```diff
@@ -0,0 +1,159 @@
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"inventory_type": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"uuid": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (l SystemCenterVirtualMachineManagerInventoryItemsDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SystemCenterVirtualMachineManager.InventoryItems
+
+			var state SystemCenterVirtualMachineManagerInventoryItemsDataSourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			scvmmServerId, err := inventoryitems.ParseVMmServerIDInsensitively(state.SystemCenterVirtualMachineManagerServerId)
+			if err != nil {
+				return err
+			}
+
+			id := fmt.Sprintf("%s/inventoryItems", scvmmServerId.ID())
+
+			resp, err := client.ListByVMMServer(ctx, *scvmmServerId)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("%s was not found", id)
+				}
+				return fmt.Errorf("reading %s: %+v", id, err)
+			}
+
+			if model := resp.Model; model != nil {
+				inventoryItems := flattenInventoryItems(model)
+				if len(inventoryItems) == 0 {
+					return fmt.Errorf("no inventory items were found for %s", scvmmServerId)
+				}
+				state.InventoryItems = inventoryItems
+			}
+
+			metadata.ResourceData.SetId(id)
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func flattenInventoryItems(input *[]inventoryitems.InventoryItem) []InventoryItem {
+	results := make([]InventoryItem, 0)
+	if input == nil {
+		return results
+	}
+
+	for _, item := range *input {
+		inventoryItem := InventoryItem{
+			id: pointer.From(item.Id),
+		}
+
+		if props := item.Properties; props != nil {
+			if v, ok := props.(inventoryitems.CloudInventoryItem); ok {
+				inventoryItem.name = pointer.From(v.InventoryItemName)
+				inventoryItem.Uuid = pointer.From(v.Uuid)
+				inventoryItem.InventoryType = string(inventoryitems.InventoryTypeCloud)
+			} else if v, ok := props.(inventoryitems.VirtualMachineInventoryItem); ok {
+				inventoryItem.name = pointer.From(v.InventoryItemName)
+				inventoryItem.Uuid = pointer.From(v.Uuid)
+				inventoryItem.InventoryType = string(inventoryitems.InventoryTypeVirtualMachine)
+			} else if v, ok := props.(inventoryitems.VirtualMachineTemplateInventoryItem); ok {
+				inventoryItem.name = pointer.From(v.InventoryItemName)
+				inventoryItem.Uuid = pointer.From(v.Uuid)
+				inventoryItem.InventoryType = string(inventoryitems.InventoryTypeVirtualMachineTemplate)
+			} else if v, ok := props.(inventoryitems.VirtualNetworkInventoryItem); ok {
+				inventoryItem.name = pointer.From(v.InventoryItemName)
+				inventoryItem.Uuid = pointer.From(v.Uuid)
+				inventoryItem.InventoryType = string(inventoryitems.InventoryTypeVirtualNetwork)
+			}
```

Comment:
Given the number of items that could be returned here, this isn't going to be particularly usable.

Perhaps it'd be sensible to add a `Name` and `Type` filter to this resource, to obtain an `Item` rather than a list of `Items`, to make this more useful?

Reply:
The Inventory Items are automatically created by the service while creating System Center Virtual Machine Manager. The end users can't create the Inventory Items. So the users don't know the name of the Inventory Item when getting an specific Inventory Item. Hence, I made this data source to get a list of Inventory Items.

Reply:
But seems it makes sense to add a filter "Inventory Type". So I updated PR to add a filter "Inventory Type" to get a list of the inventory items by the "Inventory Type".